staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -72,12 +72,16 @@ type EgressSelection struct {
 // Connection provides the configuration for a single egress selection client.
 type Connection struct {
 	// Type is the type of connection used to connect from client to konnectivity server.
-	// Currently supported values are "http-connect" and "direct".
+	// Currently supported values are "http-connect", "http-connect-uds", "grpc-uds" and "direct".
 	Type string
 
 	// httpConnect is the config needed to use http-connect to the konnectivity server.
 	// +optional
 	HTTPConnect *HTTPConnectConfig
```

Comment:
The name is misleading. It's essentially a tls config. Even if we are using grpc, as long as we are not running on uds, we still need these configs, right?

I know that we don't plan to support grpc+tcp in k8s, but I want to make this config layered correctly.